pkg/base/common/extension/registry.go

```diff
@@ -0,0 +1,21 @@
+package extension
+
+import "github.com/transaction-wg/seata-golang/pkg/base/registry"
+
+var (
+	registrys = make(map[string]func() (registry.Registry, error))
+)
+
+// SetRegistry sets the registry extension with @name
+func SetRegistry(name string, v func() (registry.Registry, error)) {
+	registrys[name] = v
```

Comment:
For global map variable update, I feel it's good to have a lock to avoid data race issue.

Reply:
hello.朋友你今钉钉群了吗，加下我钉钉号scottlewis

Reply:
@dk-lockdown  已加

Reply:
@lazlaz , it's good to use the exact way like sql.go.  The most critical thing is to avoid the dup registry. Right now, the code is able to register the same registry multiple times with different instances. Then the code becomes unpredictable ^

Reply:
It might be nitpicking, but this `SetRegistry` and `GetRegistry` are not consistent ~

```
SetRegistry(string, func())
GetRegistry(string) (Registry, error)
```

I would suggest something like:
```
SetRegistry(string, Registry)
GetRegistry(string) (Registry, error)
```

Reply:
根据你的意见进行了代码修复，请看下是否合适，进行了重复判断，函数形式没变